### DISCUSSION TITLE
Livewire.dispatch - target component

### DISCUSSION THREAD

**Discussion by davidwhicker** [livewire/livewire#7474] [1. Help]
Hi all

I am using select2 and using the .on('change') method to register the change and send it to a livewire via the Livewire.dispatch('method',{key})

The issue I have is the component I'm trying to send to exists 10 times on the page. The component is generated within a foreach loop. 

Right now the dispatch works but it's firing on all instances of the component on the page. How can I target just the component instance I'm in ?

Thank you

**Comment by WilliamPinto-Olmos** [comment]
You can access the current livewire instance. I like to work this stuff with alpinejs. 

For example if you are on livewire V3

```html
<div x-data="select2">
... your select2 
</div>

<script>
Alpine.data('select2', () => ({
select2: null,
init() {
   // Your select2 initialization
   this.select2 = $(...).select2();

   this.select2.on('change', () => {
      this.initializeCommits();
   })
},
initializeCommits() {
    Livewire.hook('commit', ({ component, commit, respond, succeed, fail }) => {
        // this will check if the updated component is the current component.
        if (this.$wire.__instance.id !== component.id) return;

        succeed(({ snapshot, effect }) => {
            // do your thing...
        })
    })
},

}));
</script>
```
That was just an example, as I didnt test it, but you get the idea. You can check the according documentation.

Livewire v3: https://livewire.laravel.com/docs/javascript#commit-hooks

**Comment by timetorock** [comment]
@FortuneV13 As I understand you have foreach cycle, just add ID of the component iteration to the event:

`Livewire.dispatch('method{{ $someComponentUniqueKey }}', {someComponentUniqueKey})`

Example with Alpine to open/close block:

`@foreach($components as $component)`
`x-on:click="$dispatch('method-{{ $component->ID() }}')"`
`x-on:method-{{ $component->ID() }}.window="open = !open"`
`@endforeach`

This will trigger event only for that component on the page.
